### ISSUE TITLE
About the accuracy

### ISSUE THREAD

**Issue by anguoyang** [deepinsight/insightface#76]
Hi all, I have tested on some datasets, and found some interesting characteristics/rules:
1.dataset1, 100 different people with 1 image for each person. the accuracy is high
2.dataset2, about 20 pairs of images, each person with 2 images, then compare image pairs of the same person, e.g compare A-1.jpg and A-2.jpg, which belong the same person, then the result is really bad.

all the images captured in the real environment by IP cameras.

**Comment by nttstar** [comment]
For dataset2, how did you compare them?

**Comment by anguoyang** [comment]
use the same method with dataset1.
1.) face detection on 2 images and align it (to size 112x112) based on the 5 key facial points, then we get aligned images
2.)extract the deep representation(512 float values) of 2 images respectively using the mxnet predict API
3.)compare the 2 vectors on cosine similarity

**Comment by anguoyang** [comment]
I would like to send the datasets to you if needed by email

**Comment by nttstar** [comment]
How did you determine the threshold?   OK, please send it to me~ Aligned images are preferred.

**Comment by anguoyang** [comment]
I have already sent it to your email(guojia@gmail.com).
I have not set the threshold, but observed the similarity values

**Comment by nttstar** [comment]
I will check it later and response to you in this issue~

**Comment by anguoyang** [comment]
Hi@nttstar, sorry for reopen this issue, I read a document by chance today:
https://hackernoon.com/building-a-facial-recognition-pipeline-with-deep-learning-in-tensorflow-66e7645015b8
"For a loss function, FaceNet uses “triplet loss”. Triplet loss relies on minimizing the distance from positive examples, while maximizing the distance from negative examples." 

Is it possible to use similar loss functions? from the testing, I found that the similarity between images of the same person always above 0.6, which is good, but the distance is not so "minimized".

**Comment by nttstar** [comment]
Triplet loss is commonly worse than sphereface/cosface/arcface by using the same training dataset.

**Comment by anguoyang** [comment]
okay, thank you

**Comment by anguoyang** [comment]
Hi@nttstar, is there any progress?

**Comment by nttstar** [comment]
@anguoyang Sorry I do not have time to work on it. But I checked the images and I believe it must be no problem to distinguish different people in this dataset as we had tested a very similar one.

**Comment by anguoyang** [comment]
yes, I agree with you that there is no problem for distinguish, how ever the distance between different people is not far,  and the same person is not so near, could you give some hints, thank you

**Comment by nttstar** [comment]
@anguoyang Can you select a threshold so that all identities can be distinguished？We can not expect the intra-class distance can be small enough such like <0.5 , and vice versa.

**Comment by anguoyang** [comment]
不好意思，可能我的英文表述有问题。我的意思是，区分度还是不错的，也就是说，很少有把A认成B的。只是相同的人的相似度不够高，而不同的人的相似度则不够低（我用的是cosine计算两个向量的相似度），不知道有没有办法优化一下，多谢！

**Comment by whu-lyh** [comment]
@anguoyang hi， may i ask you whether you find hints to minmize the distance(increase the similarity) between the same person?

**Comment by fcs203** [comment]
> 
> 
> 不好意思，可能我的英文表述有问题。我的意思是，区分度还是不错的，也就是说，很少有把A认成B的。只是相同的人的相似度不够高，而不同的人的相似度则不够低（我用的是cosine计算两个向量的相似度），不知道有没有办法优化一下，多谢！

Does that mean the TAR@FAR is not good enough， and do  you find the solution?

**Comment by ayush431** [comment]
Getting Accuracy-Flip: 1.00000+-0.00000 while training Arcface model!
can you guys help me out what's going  wrong here??
